logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationMapper.kt

```diff
@@ -1,26 +1,52 @@
 package com.wire.kalium.logic.data.conversation
 
 import com.wire.kalium.logic.data.id.IdMapper
+import com.wire.kalium.logic.data.id.TeamId
 import com.wire.kalium.network.api.conversation.ConversationResponse
-import com.wire.kalium.persistence.dao.Conversation as PersistedConversation
+import com.wire.kalium.persistence.dao.ConversationEntity as PersistedConversation
 
 interface ConversationMapper {
-    fun fromApiModel(apiModel: ConversationResponse): Conversation
-    fun fromApiModelToDaoModel(apiModel: ConversationResponse): PersistedConversation
+    fun fromApiModelToDaoModel(apiModel: ConversationResponse, selfUserTeamId: TeamId?): PersistedConversation
     fun fromDaoModel(daoModel: PersistedConversation): Conversation
 }
 
-internal class ConversationMapperImpl(private val idMapper: IdMapper, private val memberMapper: MemberMapper) : ConversationMapper {
+internal class ConversationMapperImpl(private val idMapper: IdMapper) : ConversationMapper {
 
-    override fun fromApiModel(apiModel: ConversationResponse): Conversation = Conversation(
-        idMapper.fromApiModel(apiModel.id), apiModel.name
-    )
+    override fun fromApiModelToDaoModel(apiModel: ConversationResponse, selfUserTeamId: TeamId?): PersistedConversation =
+        PersistedConversation(
+            idMapper.fromApiToDao(apiModel.id), apiModel.name, apiModel.getConversationType(selfUserTeamId), apiModel.teamId
+        )
 
     override fun fromDaoModel(daoModel: PersistedConversation): Conversation = Conversation(
-        idMapper.fromDaoModel(daoModel.id), daoModel.name
+        idMapper.fromDaoModel(daoModel.id), daoModel.name, daoModel.type.fromDaoModel(), daoModel.teamId?.let { TeamId(it) }
     )
 
-    override fun fromApiModelToDaoModel(apiModel: ConversationResponse): PersistedConversation = PersistedConversation(
-        idMapper.fromApiToDao(apiModel.id), apiModel.name
-    )
+    private fun PersistedConversation.Type.fromDaoModel(): Conversation.Type = when (this) {
+        PersistedConversation.Type.SELF -> Conversation.Type.SELF
+        PersistedConversation.Type.ONE_ON_ONE -> Conversation.Type.ONE_ON_ONE
+        PersistedConversation.Type.GROUP -> Conversation.Type.GROUP
+    }
```

Comment:
`persistence` module doesn't have access to `logic` classes. So... no :/

I was just talking with @typfel about these things. Maybe we could have a central `core-models` module or similar, to hold these data types. 

And then, the mappings `fromCoreModelToNetworkDTO` and `fromCoreModelToPersistenceEntity`could be done in respective module, instead of having them all in `logic`.